### PR TITLE
Ch test resources endpoint 159869003

### PR DESCRIPTION
**Pull request by Areo6**
@leni1
@araaliFarooq
@oxenprogrammer

### CODE REVIEW

File: app/api/models.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from flask import jsonify
+from flask import jsonify, abort
```
Comment (leni1) [inline app/api/models.py]: @Eubule I would encourage you not to use `jsonify` directly in your models but instead return a response compatible with the JSON format to the `views` module you have.

**Comment by leni1** [comment]
@Eubule Apart from the above, your code looks good to me. I would encourage you to widen test coverage for situations other than an empty field.

File: app/api/models.py (hunk 1 above)
Comment (Eubule) [reply app/api/models.py]: @leni1 thank u for your insight....jsut one question, why do u think it's not apropriate to use jsonify in models?

Comment (leni1) [reply app/api/models.py]: @Eubule on closer inspection your models seem to be behaving as views. Your views should be passing data and returning it. The checks and formatting should (mostly) happen in your models. So `jsonify`, `flask` and `abort` should appear in your controllers (views) and your models should be doing the heavy lifting of entering and returning the correct data. Hope that clarifies it for you. :smile:

**Pull request merged**